private/protocol/json/jsonutil/unmarshal.go

```diff
@@ -263,7 +264,9 @@ func (u unmarshaler) unmarshalScalar(value reflect.Value, data interface{}, tag
 			value.Set(reflect.ValueOf(&d))
 		case *time.Time:
 			// Time unmarshaled from a float64 can only be epoch seconds
-			t := time.Unix(int64(d), 0).UTC()
+			seconds, frac := math.Modf(d)
+			ms := math.Round(frac * math.Pow10(3)) // Convert fraction to milliseconds and the round the value
+			t := time.Unix(int64(seconds), int64(ms*math.Pow10(6))).UTC()
```

Comment:
Does this need to be ported to v2, or does the smithy clients handle millisecond timestamps by default?

Reply:
Will cut a follow-up item to track fixing V2 to follow the same pattern. Currently it allows up to nano-second precision.